### PR TITLE
Google summer of code

### LINKED ISSUE #228: Convert Contributing to Mautic section to RST [multiple sub-issues]
We are in the process of moving this repository to Read the Docs, and to do that we need to convert our pages from Markdown to RST. This issue is to transfer the Contribute to Mautic section: [https://contribute.mautic.org/contributing-to-mautic](https://contribute.mautic.org/contributing-to-mautic). You can also update and improve the content on these pages if you like - do check with the relevant team leads to ensure it's up to date and relevant.

## Sections to migrate 
As this is a large section I recommend that we take it one section at a time. Drop a message in the comments to be assigned to a section.

- [ ] Designer
- [ ] Developer
- [ ] Marketer
- [ ] Tester
- [ ] Translator
- [ ] Writing for Mautic
- [ ] Event organiser
- [ ] Web developer
- [ ] Server administrator
- [ ] Contributing financially
- [ ] Google Summer of Code @Iqmaa 

You can do this in the current repo, as we've got a folder set up, `/docs`, which will be used for Read The Docs.

You can build this documentation locally by cloning this repository, and following these steps: https://github.com/mautic/user-documentation/tree/5.x?tab=readme-ov-file#build-documentation-locally. Change into the /docs folder and then (assuming you have DDEV and Docker/Colima installed) type `ddev config` and select `build/html `as the root folder when configuring DDEV. Then when complete, type `ddev start` and it'll give you the link to your local testing environment.

If you need to make a link file to add an external URL, please use the command line tool 'make link'. This will generate a reference macro and a link file for you.

For internal linking check the docs on using the [:ref: role](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#the-ref-role) for linking within the same document, and using [:doc: to link across pages](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#the-doc-role).

**Please note: if you see folders called .revs these are old version control files from Grav. Ignore them! (they should be deleted) and instead work on the actual files.**

When you've converted the file/s, **they need to live within the /docs folder**. We can create a structure using sub-folders to organise the sections. You also need to add the pages to the Table of Contents in the index.rst file for them to show on the menu. Docs on the ToC can be found [here](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/indices.html).

We highly recommend that you use VS Code with Vale installed. Let us know if you have any problems.

Some useful resources: 
- https://sublime-and-sphinx-guide.readthedocs.io/en/latest guide to Sphinx/RST formatting
- https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#use-the-external-links-extension how to do external links - always check the /docs/links directory first! You can use the terminal command '`make link`' to create the link file automagically.

There are many MD to RST tools out there which can help speed up the process, however you will still need to review your output to ensure it passes our Vale grammar rules. Headings must always be unique, as well.

We use the following for headings:

```
H1 ###
H2 ***
H3 ===
H4 ---
H5 ~~~
```

Please ask in #t-education on Slack if you get stuck with anything!

### PR DESCRIPTION
**Pull request by Iqmaa**
Fixes #228 subsection Google Summer of Code. Converted the Google summer of Code page to RST following the given guidelines

### CODE REVIEW

File: docs/contributing/google_summer_of_code.rst [hunk 1]
```diff
@@ -0,0 +1,160 @@
+.. image:: images/Summer_of_Code.png
+   :width: 600
+
+Google Summer of Code at Mautic
+###############################
+
+This page contains information about Mautic's participation in Google Summer of Code, often abbreviated to GSoC. Please feel free to contact us via Slack below for clarifications and more information.
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:7]:
```suggestion
This page contains information about Mautic's participation in Google Summer of Code, often abbreviated to GSoC. Please feel free to contact the team via Slack below for clarifications and more information.
```

File: docs/contributing/google_summer_of_code.rst [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+What can I do today?
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:9]:
```suggestion
What to get started with today?
```

File: docs/contributing/google_summer_of_code.rst [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+********************
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:10]: Also fix the underline ;)

File: docs/contributing/google_summer_of_code.rst [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+Today you should read through the ideas for projects, and start working on your application. Get familiar with Mautic, make some contributions so you're familiar with the workflow, and get to know the community. Join the :xref:`Mautic Community Slack` channel - #gsoc-discuss - and post any questions you have there, after reading this page fully and also reviewing the project proposals.
+
+Student Information
+*******************
+
+Do you want to contribute to one of the world's first Open Source Marketing Automation platform? GSoC could be your chance. Please refer to the information below.
+
+If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling document which we maintain year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an “I'm interested” email or Slack message - please respect their time.
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:19]:
```suggestion
If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling submission process which is maintained year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an “I'm interested” email or Slack message - please respect their time.
```

File: docs/contributing/google_summer_of_code.rst [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+Student Responsibilities
+************************
+
+You are the key ingredient for your project's success. This project is important to you, your mentor, and the entire Mautic Community.
+
+Your responsibilities include:
+
+- Communicating early and often with your mentor.
+- Updating every week about what you're learning, how you're doing, challenges and successes. This is key way to keep the entire Mautic Community informed.
+- Working with your mentor on realistic achievable milestones that provide for regular deliverables and feedback.
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:58]:
```suggestion
- Working with your mentor on realistic achievable milestones that provide for regular deliverable targets and feedback.
```

File: docs/contributing/google_summer_of_code.rst [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+Student Application
+*******************
+
+Please read and follow the student application process below.
+
+Google Summer of Code Application Process
+=========================================
+
+Interested in applying for Google Summer of Code with Mautic? Here is how you do that!
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:70]:
```suggestion
Interested in applying for Google Summer of Code with Mautic? Here is how you do that.
```

File: docs/contributing/google_summer_of_code.rst [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+Apply
+~~~~~
+
+Student applications open on March 18 2024 - 18:00 UTC.
+
+The application process consists of 3 steps:
+
+1. Create an account on the :xref:`Mautic Community Forums`
+2. Join the #gsoc-discuss and #t-product channels on Slack
+3. Submit the application/proposal including all requirements in the next section at the Google Summer of Code Site. We **strongly** encourage you to not submit your final PDF until you get feedback about the proposal from your mentor(s).
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:114]:
```suggestion
3. Submit the application/proposal including all requirements in the next section at the Google Summer of Code Site. We **strongly** encourage you to not submit your final PDF until you get feedback about the proposal from your mentors.
```

File: docs/contributing/google_summer_of_code.rst [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+Help is available! Share your application in progress and get feedback. The mentors can help verify your understanding of the project and provide guidance on whether your plan is good.
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:116]:
```suggestion
Help is available. Share your application in progress and get feedback. The mentors can help verify your understanding of the project and provide guidance on whether your plan is good.
```

File: docs/contributing/google_summer_of_code.rst [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+Google Summer of Code Application and Supplemental Material Requirements
+------------------------------------------------------------------------
+
+In addition to the information Google requests, the Mautic Project requires you to submit the following information:
+
+- What project are you applying to? Seriously consider choosing a project for proposal from ideas list. While it is possible to propose your own project, it is often more successful for you to select one that is already proposed by a mentor.
+- Contact Information - List your Name, Email Address, Blog URL, and any additional contact information you think we need. We tend to email so be sure you're reading your email.
+- About You - Consider these questions
+
+  - Do you have any past involvement with the Mautic Project or another Open Source project as a contributor? What kind of contributions have you made (code and non-code)? Please share any public contributions you have made (urls, etc.)
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:131]:
```suggestion
  - Do you have any past involvement with the Mautic Project or another Open Source project as a contributor? What kind of contributions have you made (code and non-code)? Please share any public contributions you have made (URLs, etc.)
```

File: docs/contributing/google_summer_of_code.rst [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+  - What are you studying? Why?
+  - What motivates you to do your best? How do you stay focused? Briefly describe how you organize your work.
+
+- Why Us? - Consider these questions
+
+  - Why do you want to work with the Mautic Project?
+  - Do you plan to continue contributing to the Mautic Project after GSoC? If yes, what project(s) are you interested with?
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:138]:
```suggestion
  - Do you plan to continue contributing to the Mautic Project after GSoC? If yes, what projects are you interested with?
```

File: docs/contributing/google_summer_of_code.rst [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+
+- Why This Project? - Consider these questions
+
+  - Which specific project interests you? If you are proposing your own project, do it here. If you are proposing the project, explain why it benefits the community.
+  - Have you contacted the mentor(s) listed for the project?
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:143]:
```suggestion
  - Have you contacted the mentors listed for the project?
```

File: docs/contributing/google_summer_of_code.rst [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+  - What relevant experience do you have for this project?
+  - What do you want to gain from this project?
+  - Describe this project in your own words. What does it mean to you? What do you think the final deliverable is?
+
+- Why should we choose you over other applicants?
+- Propose a rough schedule for this project. Ideally you will have regular deliverables throughout the entire period.
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:149]:
```suggestion
- Propose a rough schedule for this project. Ideally you'll have regular deliverable tasks throughout the entire period.
```

File: docs/contributing/google_summer_of_code.rst [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+- What are you specific plans from May to September (eg: Exams in university, internships, vacations..etc). We need to know if there are periods of time where you are likely to be unavailable. We are better at accommodating these periods if we know of them in advance.
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:150]:
```suggestion
- What are you specific plans from May to September (for example, exams in university, internships, vacations, etc). We need to know if there are periods of time where you are likely to be unavailable. We are better at accommodating these periods if we know of them in advance.
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Great job on this so far @Iqmaa - I have made some suggestions for you to fix the reported Vale errors.

File: docs/contributing/google_summer_of_code.rst [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+*******************************
+
+Today you should read through the ideas for projects, and start working on your application. Get familiar with Mautic, make some contributions so you're familiar with the workflow, and get to know the community. Join the :xref:`Mautic Community Slack` channel - #gsoc-discuss - and post any questions you have there, after reading this page fully and also reviewing the project proposals.
+
+Student Information
+*******************
+
+Do you want to contribute to one of the world's first Open Source Marketing Automation platform? GSoC could be your chance. Please refer to the information below.
+
+If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling submission process which is maintained year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an “I'm interested” email or Slack message - please respect their time.
```
Comment (RCheesley) [inline docs/contributing/google_summer_of_code.rst:19]:
```suggestion
If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling submission process which Mautic maintains year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an ``“I'm interested”`` email or Slack message - please respect their time.
```

Vale fix

**Review by RCheesley — CHANGES_REQUESTED** [review]
Just some small changes suggested by Vale - shout if you need any help!

**Comment by Iqmaa** [comment]
> Just some small changes suggested by Vale - shout if you need any help!

Done,
Thanks @RCheesley !

**Review by RCheesley — DISMISSED** [review]
Great job @Iqmaa - this is good to go 🚀

**Comment by RCheesley** [comment]
This is good to merge now, the other issues causing it to fail are due to #244.

**Pull request merged**